### PR TITLE
New Module: humid

### LINKED ISSUE #5484: new module: HUMID
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by Joon-Klaps**
## PR checklist

Closes #5484 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [X] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/humid/meta.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+tools:
+  - "humid":
+      description: "HUMID -- High-performance UMI Deduplicator"
+      homepage: "https://github.com/jfjlaros/HUMID"
+      documentation: "https://humid.readthedocs.io/en/latest/"
+      tool_dev_url: "https://github.com/jfjlaros/HUMID"
+      licence: ["MIT"]
+
+input:
+  # Only when we have meta
```
Comment (matthdsm) [inline modules/nf-core/humid/meta.yml:20]: Remove the comment line

File: modules/nf-core/humid/meta.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1', single_end:false ]`
+  - reads:
+      type: file
+      description: Fastq file(s) to deduplicate
+      pattern: "*.{fastq,fq,fastq.gz,fq.gz}"
+  - umis:
```
Comment (matthdsm) [inline modules/nf-core/humid/meta.yml:30]: Add meta2 and make sure the name here reflects the name in the module

File: modules/nf-core/humid/meta.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      type: file
+      description: Annotated Fastq file(s)
+      pattern: "*.{fastq,fq,fastq.gz,fq.gz}"
+
+  - stats:
+      type: directory
+      description: Directory containg statistics file, use for multiqc.
+      pattern: "${prefix}/"
+
+  - versions:
```
Comment (matthdsm) [inline modules/nf-core/humid/meta.yml:57]: Also add the log output

**Review by matthdsm — CHANGES_REQUESTED** [review]
Some comments, looks good otherwise

**Comment by Joon-Klaps** [comment]
@matthdsm, had to modify tests a little bit, checksum on the log was inconsistent. I'm guessing it's inconsistent as the log file also includes computation time.

**Pull request merged**